File: _old/LogiX/Circuits/Logic/XORGateLogic.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace LogiX.Circuits.Logic
{
    class XORGateLogic : IGateLogic
    {
        public int GetExpectedInputAmount()
        {
            return 2;
        }

        public LogicValue GetOutput(LogicValue[] inputs)
        {
            LogicValue a = inputs[0];
            LogicValue b = inputs[1];

            if (a == LogicValue.NAN || b == LogicValue.NAN)
            {
                return LogicValue.NAN;
            }

            if((a == LogicValue.HIGH && b == LogicValue.LOW) || (a == LogicValue.LOW && b == LogicValue.HIGH))
            {
                return LogicValue.HIGH;
            }
            else
            {
                return LogicValue.LOW;
            }
        }
    }
}

File: src/gate-logic/xor_gate_logic.cpp
#include "gate-logic/xor_gate_logic.hpp"
#include "circuits/circuit_input.hpp"
#include <vector>

LogicValue XORGateLogic::PerformGateLogic(std::vector<CircuitInput*> inputs) {
    int high = 0;
    for (int i = 0; i < inputs.size(); i++) {
        CircuitInput* inp = inputs.at(i);

        for (int j = 0; j < inp->bits; j++) {
            if (inp->GetValue(j) == LogicValue_HIGH) {
                high = high + 1;
            }
        }
    }

    return high % 2 == 0 ? LogicValue_LOW : LogicValue_HIGH;
}

const char* XORGateLogic::GetLogicName() {
    return "XOR";
}